### PR TITLE
Chris tests

### LINKED ISSUE #7: Testing GamePage
- [ ] Write a test for all possible render conditions (list them here in comments)
- [ ] Pass them

### LINKED ISSUE #8: Change Test Sentence in WordSelector
Change line 15 in WordSelector.test to test 'if isGenerator is **not** true'

### LINKED ISSUE #33: Test Wordselector

### PR DESCRIPTION
**Pull request by Chriscastanuela**
## Description
* Tests for `Gamepage` and `Wordselector` components

## Affected areas of application
* `Gamepage` and `Wordselector` components

## Screenshots (if applicable):
Screen Shot 2020-10-28 at 7 29 04 PM

## How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration
- [x] Test A - Jest

## Relevant Tickets
* Closes #7 
* Closes #33
Closes #8

### CODE REVIEW

**Review by GreyMatteOr — COMMENTED** [review]
🔥

File: src/Gamepage/Gamepage.js [hunk 1]
```diff
@@ -26,14 +26,20 @@ export class Gamepage extends Component {
   }
 
   mapAttempts = () => {
+    let theKey = 0;
     return this.props.attempts.map(attempt => {
-      return <p>{attempt}</p>
+      theKey += 1;
+      return <p key={theKey}>{attempt}</p>
```
[the remark this answers is not in the document]
Comment (holladayian) [reply src/Gamepage/Gamepage.js:32]: Lol yeah, thanks for putting this in there!

File: src/WordSelector/WordSelector.js [hunk 2]
```diff
@@ -28,11 +28,12 @@ export class WordSelector extends Component {
           <h2>Type a word for your opponent to guess!</h2>
           <form>
             <label>
-              <input type="text" name="name" onChange={this.updateChange} />
+              <input type="text" name="name" onChange={this.updateChange} placeholder="Name"/>
```
Comment (holladayian) [inline src/WordSelector/WordSelector.js:31]: Nice use of placeholder here! I was thinking about it, the onClick in 37 below might need to get changed onSubmit (I think that's what it's called). That way users can hit "enter"/"return" to submit this as well.

File: src/WordSelector/WordSelector.test.js [hunk 3]
```diff
@@ -4,10 +4,19 @@ import { screen, render, waitFor } from '@testing-library/react';
...
+    it('should render the default heading if the `isGenerator` prop is true', () => {
 
-    render(<WordSelector />);
+      render(<WordSelector isGenerator={true}/>);
+  
+      expect(screen.getByText('Type a word for your opponent to guess!')).toBeInTheDocument();
+    })
 
-    expect(screen.getByText('Type a word for your opponent to guess!')).toBeInTheDocument();
+    it('should not render the default heading if the `isGenerator` prop is true', () => {
```
Comment (holladayian) [inline src/WordSelector/WordSelector.test.js]: ...if the `isGenerator` prop is **false**'.

File: src/WordSelector/WordSelector.test.js [hunk 4]
```diff
@@ -4,10 +4,19 @@ import { screen, render, waitFor } from '@testing-library/react';
...
+
+      render(<WordSelector />);
```
Comment (holladayian) [inline src/WordSelector/WordSelector.test.js:17]: Also, I know it renders the same thing, but I think for dev empathy we should still pass the prop of "isGenerator={false}" into here.

**Review by holladayian — CHANGES_REQUESTED** [review]
I added this as a "request change" just so we can change it before it get's lost in the merge. Should just be changing "prop is true" to "prop is false" on line 15.

File: src/Gamepage/Gamepage.test.js [hunk 5]
```diff
@@ -1,16 +1,122 @@
...
+      makeGuess={
+        async (newGuess) => {
+          // const guess = await apiCalls.makeGuess(newGuess)
+          // this.setState({guess})
+          console.log('newGuess', newGuess)
+        }
+      }
+      />);
+      // expect(screen.getAllByTestId('splitDisplay').toBeInTheDocument)
+      expect(screen.getByTestId('1')).toBeInTheDocument()
```
[the remark this answers is not in the document]
Comment (holladayian) [reply src/Gamepage/Gamepage.test.js:94]: damn

**Review by holladayian — COMMENTED** [review]
Disregard request changes. It looks like it got cleaned up in a latter commit!

**Review by holladayian — APPROVED** [review]

**Review by holladayian — APPROVED** [review]
I had added a request for changes but I didn't see that what I was requesting got cleaned up on a latter commit. Just wanted to be thorough and approve this!

**Comment by holladayian** [comment]
Looks good! Will merge! Also I added an issue to close (#8) just an fyi.

**Pull request merged**